src/src/PluginStructs/P099_data_struct.h

```diff
@@ -0,0 +1,98 @@
+#ifndef PLUGINSTRUCTS_P099_DATA_STRUCT_H
+#define PLUGINSTRUCTS_P099_DATA_STRUCT_H
+
+#include "../../_Plugin_Helper.h"
+#include "../../ESPEasy_common.h"
+
+#ifdef USES_P099
+
+#include <XPT2046_Touchscreen.h>
+
+// #define PLUGIN_099_DEBUG    // Additional debugging information
+
+// Define default values for both ESP32/lolin32 and D1 Mini
+#ifdef ESP32
+  #define P099_TS_CS 12
+#else // ESP8266/ESP8285
+  #define P099_TS_CS  0 // D3
+#endif // ESP32
+
+#define P099_TS_TRESHOLD        15    // Treshold before the value is registered as a proper touch
+#define P099_TS_ROTATION        2     // Rotation 0-3 = 180/270/0/90 degrees, should be set to the same rotation degrees, not 0-3 value!, as the screen it is mounted on
+#define P099_TS_SEND_XY         true  // Enable X/Y events
+#define P099_TS_SEND_Z          false // Disable Z events
+#define P099_TS_SEND_OBJECTNAME true  // Enable objectname events
+#define P099_TS_USE_CALIBRATION false // Disable calibration
+#define P099_TS_LOG_CALIBRATION true  // Enable calibration logging
+#define P099_TS_X_RES           240   // Pixels, should match with the screen it is mounted on
+#define P099_TS_Y_RES           320
+#define P099_INIT_OBJECTCOUNT   8     // Initial setting
+
+#define P099_TOUCH_X_INVALID  4095 // When picking up spurious noise (or an open/not connected TS-CS pin), these are the values that turn up
+#define P099_TOUCH_Y_INVALID  4095
+#define P099_TOUCH_Z_INVALID  255
+
+#define P099_MaxObjectNameLength 16 // 15 character objectnames + terminating 0
+#define P099_MaxObjectCount      40 // This count of touchobjects should be enough, because of limited settings storage, 960 bytes + 8 bytes calibration coordinates
+
+// The setting structures
+typedef struct
+{
+  uint16_t x = 0;
+  uint16_t y = 0;
+} tP099_Point;
+
+typedef struct
+{
+  char        objectname[P099_MaxObjectNameLength] = { 0 };
+  tP099_Point top_left;
+  tP099_Point bottom_right;
+} tP099_Touchobjects;
+
+typedef struct {
+  tP099_Point top_left;
+  tP099_Point bottom_right;
+} tP099_Calibration;

```

Comment:
These are not members of the data_struct object, which means they are always present in the global scope.
Regardless whether the plugin is active or not and thus will also be shared among multiple instances of this plugin.

If you need to have a struct for saving, then make a separate struct for it and declare that as a member of the data_struct object.
When declared like this `typedef struct { ... } var_name;` you define and declare the variable and since it is outside the scope of a struct or class, it is a variable in the global scope.